### PR TITLE
Add company model

### LINKED ISSUE #35: Add Company model
Company should has:
- users
- articles
- roles

### PR DESCRIPTION
**Pull request by RasimKhusaenov**
### Summary

- Added Company model
- Added CompanyMember model (for company users and their roles)
- Added user and company FKs to article (for company articles)
- Added ProgressBar

### How it works

N/A

### Test plan

N/A

### Deploy notes

- Run `bin/rails db:migrate`
- Run
```
def set_user_to_articles
  relation = Article.where(user_id: nil)
  default_user = User.find_by(role: :superadmin)

  progress_bar = ProgressBar.create(total: relation.count, format: "%t: |%B %P %e|")

  relation.find_each do |element|
    element.update!(user: default_user)
    progress_bar.increment
  end
end
```

### Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [x] My pull request is shorter than 500 lines

### References
- Resolves #35 Issue

### CODE REVIEW

File: db/migrate/20220527201132_add_author_and_company_to_articles.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddAuthorAndCompanyToArticles < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :articles, :user, default: User.where(role: :superadmin).first.id, null: false
```
Comment (KilinMikhail) [inline db/migrate/20220527201132_add_author_and_company_to_articles.rb:3]: I think it is strange to set default user id on the database level. If SuperAdmin id will change, You will need to create new migration. Moreover, in you schema you can see id: 1 as default value. So if i will try to setup application on my own computer, there will not this user when I will load schema, so, you need to setup this value on app level if it is needed.

Comment (RasimKhusaenov) [reply db/migrate/20220527201132_add_author_and_company_to_articles.rb:3]: I understood my mistake, perhaps. I should set default value for existing records by executing command after migration. Am I right?

Comment (KilinMikhail) [reply db/migrate/20220527201132_add_author_and_company_to_articles.rb:3]: I think this value should not be set on database level. As I think, if you need to set this value for previous values in database, you should make migration with null: true. Then you need update old fields by deploy script. And after that you need to create new migration taht will change constraint to null: false

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
 
   TOKEN_EXPIRATION_TIME = 15.minutes
 
+  has_many :company_members, dependent: :destroy
+  has_many :articles, dependent: :nullify
```
Comment (RamilyaNigmatullina) [inline app/models/user.rb:7]: I see that `user_id` is required in the article model (`belongs_to :user`). But it seems the value can become empty if we delete the user.

File: db/migrate/20220527194241_create_companies.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateCompanies < ActiveRecord::Migration[6.1]
+  def change
+    create_table :companies do |t|
+      t.string :official_name, null: false
```
Comment (RamilyaNigmatullina) [inline db/migrate/20220527194241_create_companies.rb:4]: Do we need to add validation for this filed in the Company model?

File: app/models/company.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Company < ApplicationRecord
+  extend Enumerize
```
Comment (KilinMikhail) [inline app/models/company.rb:2]: ?

**Review by KilinMikhail — APPROVED** [review]

File: db/migrate/20220527201132_add_author_and_company_to_articles.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddAuthorAndCompanyToArticles < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :articles, :user, foreign_key: true
```
Comment (RamilyaNigmatullina) [inline db/migrate/20220527201132_add_author_and_company_to_articles.rb:3]: Do we need to add null false constraint?

Comment (RasimKhusaenov) [reply db/migrate/20220527201132_add_author_and_company_to_articles.rb:3]: Mikhail and I discussed this. We have articles in the prod that do not have an author. As part of this PR, we want to migrate it, assign an author to these articles and add a constraint next time.

**Review by RamilyaNigmatullina — APPROVED** [review]

**Pull request merged**